### DISCUSSION TITLE
Results Scrollback on MacOS

### DISCUSSION THREAD

**Discussion by vrmer** [inducer/pudb#639] [Troubleshooting] [answered]
I have been an avid user of PUDB for a little over a year now, and while I used to be able to go to the results scrollback in the command-line using the arrow keys, now I can only scroll through previous inputs. Is there a way to enter the results scrollback? Shift+arrow doesn't work for me.

**Comment by inducer** [comment, marked as answer]
This was changed in #600, see also the discussion there. One way to get to the scrollback is to go "Cursor Right" to get to the "Clear" button and go up from there. We *could* make a specific hotkey if you think that'd be better.

**Reply by vrmer** [reply]
Ah I see, I saw #600 but didn't find the way to solve this. The "Clear" to up hack works for me, thanks! Maybe this could be added to the pudb help, but it is not crucial.

Thanks again!

**Reply by inducer** [reply]
I was looking into adding a note about this to the help screen, and I discovered that Shift+PageUp/Down are documented as a way to scroll through the shell output, and that seems to work for me.

**Reply by vrmer** [reply]
I saw but on a MacBook, PageUp and PageDown can be only reached via the Fn+Arrow key combination, and that somehow doesn't translate into the terminal.

**Reply by inducer** [reply]
Interesting. If you can find a reasonable key combination that happens to work on a Mac, I'd be happy to consider a PR that adds this as an alternative.

**Reply by vrmer** [reply]
Oh that'd be really great! Would Shift+Up/Down arrow work? I think that's what the documentation lists anyway if I remember correctly, but it seems to do the same as the normal arrow keys.

**Reply by inducer** [reply]
It lists Shift+PageUp/Down:

https://github.com/inducer/pudb/blob/74a855d9e3a2c1ee9787052cd49c2b65518d67ad/pudb/debugger.py#L100

I guess I could live with Shift+Up/Down, though that does clash with the CUA keys for "extend selection down".

**Reply by asmeurer** [reply]
Are Control-Up or Meta-Up options?

**Reply by vrmer** [reply]
I think Meta-Up might work!

**Reply by inducer** [reply]
Once you all figure out what works on Mac, please find someone to file a PR.
